tests/test_subscription.py

```diff
@@ -478,6 +487,22 @@ def test_run_transmogrifier_did_type(self, vo, rse_factory, rucio_client, root_a
         rules = [rule for rule in rucio_client.list_did_rules(scope=tmp_scope.external, name=dsn) if str(rule['subscription_id']) == str(subid)]
         assert len(rules) == 2
 
+    def test_deactivate_subscription(self, rse_factory, rucio_client, vo):
+        """SUBSCRIPTION (CLIENT): Verify a subscription can be marked as inactive"""
+        subscription_name = uuid()
+        rse1, _ = rse_factory.make_mock_rse()
+        rse2, _ = rse_factory.make_mock_rse()
+        rse_expression = '%s|%s' % (rse1, rse2)
+        account_name = uuid()[:10]
+        add_account(InternalAccount(account_name, vo=vo), AccountType.USER, 'rucio@email.com')
+        rucio_client.add_subscription(
+            name=subscription_name, account=account_name, filter_={'project': self.projects, 'datatype': ['AOD', ], 'excluded_pattern': self.pattern1, 'account': ['tier0', ]},
+            replication_rules=[{'rse_expression': rse_expression, 'copies': 2, 'activity': self.activity}], lifetime=100000, retroactive=False, dry_run=False, comments='Ni ! Ni!')
+
+        rucio_client.deactivate_subscription(name=subscription_name, account=account_name)
+        states = [r['state'] for r in rucio_client.list_subscriptions(name=subscription_name, account=account_name)]
+        assert states[0] == 'INACTIVE'
```

Comment:
Similarly here, we should add an earlier assertion that ensures that the state was not `INACTIVE` before, to actually verify that this update action is changing the state.